pkg/generate/code/set_resource.go

```diff
@@ -908,40 +922,54 @@ func SetResourceIdentifiers(
 		}
 
 		if r.IsPrimaryARNField(memberName) {
-			// r.ko.Status.ACKResourceMetadata.ARN = identifier.ARN
-			arnOut += fmt.Sprintf(
-				"\n%s%s.Status.ACKResourceMetadata.ARN = %s.ARN\n",
-				indent, targetVarName, sourceVarName,
-			)
 			continue
-
 		}
+
 		// Check that the field has potentially been renamed
 		renamedName, _ := r.InputFieldRename(
 			op.Name, memberName,
 		)
 
 		isPrimaryIdentifier := memberName == primaryIdentifier
 		cleanMemberNames := names.New(renamedName)
-		cleanMemberName := cleanMemberNames.Camel
 
 		memberPath := ""
-		_, inSpec := r.SpecFields[renamedName]
-		_, inStatus := r.StatusFields[renamedName]
+		var targetField *model.Field
+
+		specField, inSpec := r.SpecFields[renamedName]
+		statusField, inStatus := r.StatusFields[renamedName]
 		switch {
 		case inSpec:
 			memberPath = cfg.PrefixConfig.SpecField
+			targetField = specField
 		case inStatus:
 			memberPath = cfg.PrefixConfig.StatusField
+			targetField = statusField
 		case isPrimaryIdentifier:
 			panic("Primary identifier field '" + memberName + "' in operation '" + op.Name + "' cannot be found in either spec or status.")
 		default:
 			continue
 		}
 
+		targetVarPath := fmt.Sprintf("%s%s", targetVarName, memberPath)
```

Comment:
Could use [getSanitizedMemberPath](https://github.com/aws-controllers-k8s/code-generator/blob/93f829bb5b5420a20cb1af70a0c562eb61402f2b/pkg/generate/code/check.go#L231) here

Reply:
Hmm, I think there would still be roughly the same number of LoC even if I use `getSanitizedMemberPath`, here. Mainly because I need a reference to the `*Field` - as its required for the `setResourceForScalar` parameters. Although I'm essentially doing the same logic, just with an extra return value. 

I might refactor `getSanitizedMemberPath` to return the field as well, but not in this PR.